streams/src/main/java/org/apache/kafka/streams/state/internals/VersionedKeyValueToBytesStoreAdapter.java

```diff
@@ -0,0 +1,196 @@
+        return serializeAsBytes(versionedRecord);
+    }
+
+    @Override
+    public String name() {
+        return inner.name();
+    }
+
+    @Deprecated
+    @Override
+    public void init(final ProcessorContext context, final StateStore root) {
+        inner.init(context, root);
+    }
+
+    @Override
+    public void init(final StateStoreContext context, final StateStore root) {
+        inner.init(context, root);
+    }
+
+    @Override
+    public void flush() {
+        inner.flush();
+    }
+
+    @Override
+    public void close() {
+        inner.close();
+    }
+
+    @Override
+    public boolean persistent() {
+        return inner.persistent();
+    }
+
+    @Override
+    public boolean isOpen() {
+        return inner.persistent();
+    }
+
+    @Override
+    public <R> QueryResult<R> query(final Query<R> query, final PositionBound positionBound, final QueryConfig config) {
+        return inner.query(query, positionBound, config);
+    }
+
+    @Override
+    public Position getPosition() {
+        return inner.getPosition();
+    }
+
+    @Override
+    public byte[] putIfAbsent(final Bytes key, final byte[] value) {
+        throw new UnsupportedOperationException("Versioned key-value stores do not support putIfAbsent(key, value)");
+    }
+
+    @Override
+    public void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
+        throw new UnsupportedOperationException("Versioned key-value stores do not support putAll(entries)");
+    }
+
+    @Override
+    public byte[] delete(final Bytes key) {
+        throw new UnsupportedOperationException("Versioned key-value stores do not support delete(key)");
+    }
+
+    @Override
+    public KeyValueIterator<Bytes, byte[]> range(final Bytes from, final Bytes to) {
+        throw new UnsupportedOperationException("Versioned key-value stores do not support range(from, to)");
+    }
+
+    @Override
+    public KeyValueIterator<Bytes, byte[]> reverseRange(final Bytes from, final Bytes to) {
+        throw new UnsupportedOperationException("Versioned key-value stores do not support reverseRange(from, to)");
+    }
+
+    @Override
+    public KeyValueIterator<Bytes, byte[]> all() {
+        throw new UnsupportedOperationException("Versioned key-value stores do not support all()");
+    }
+
+    @Override
+    public KeyValueIterator<Bytes, byte[]> reverseAll() {
+        throw new UnsupportedOperationException("Versioned key-value stores do not support reverseAll()");
+    }
+
+    @Override
+    public <PS extends Serializer<P>, P> KeyValueIterator<Bytes, byte[]> prefixScan(final P prefix, final PS prefixKeySerializer) {
+        throw new UnsupportedOperationException("Versioned key-value stores do not support prefixScan(prefix, prefixKeySerializer)");
+    }
+
+    @Override
+    public long approximateNumEntries() {
+        throw new UnsupportedOperationException("Versioned key-value stores do not support approximateNumEntries()");
```

Comment:
Just wondering if we could (should?) return the "latest value store" number here?

Reply:
We can definitely discuss whether we want to expose this for the RocksDB-based implementation but I think it'll require adding the method (perhaps with a default implementation which throws UnsupportOperationException) into the VersionedKeyValueStore interface, since this adapter here is general-purpose and won't be able to adapt from a generic VersionedKeyValueStore implementation without the method being part of the interface. In light of that, let's defer to future discussions for a follow-up KIP since there are other methods we might like to add to the interface in the future as well.